src/main/java/com/commercetools/sync/products/helpers/ProductReferenceResolver.java

```diff
@@ -432,6 +432,6 @@ public ProductReferenceResolver(@Nonnull final ProductSyncOptions productSyncOpt
         }
 
         return collectionOfFuturesToFutureOfCollection(futures, toList())
-            .thenApply(maps -> productKeys.isEmpty() ? Collections.emptyMap() : maps.get(0));
+            .thenCompose(ignored -> productService.cacheKeysToIds(Collections.emptySet())); // return all cache.
```

Comment:
also naming is confusing. `productService.cacheKeysToIds` when the user ask nothing, it shouldn't return everything it has.`getKeysToIdsCache` would be good candidate name for the new method

Reply:
Same here, maybe I could add those points to a separate GitHub issue, soon we will describe the new concept for the cache implementation,  https://github.com/commercetools/commercetools-sync-java/pull/581#discussion_r496616879

Reply:
Added as a context: https://github.com/commercetools/commercetools-sync-java/issues/582